app/controllers/posts_controller.rb

```diff
@@ -91,10 +91,12 @@ def unhide
   end
 
   def new
-    @post = Post.new(character: current_user.active_character, user: current_user)
+    @post = Post.new(user: current_user)
     @post.board_id = params[:board_id]
     @post.section_id = params[:section_id]
-    @post.icon_id = (current_user.active_character ? current_user.active_character.default_icon.try(:id) : current_user.avatar_id)
+    @post.written = @post.replies.new(user: current_user, post_id: @post.id, reply_order: 0)
```

Comment:
Can we not use `@post.written.build`, did we try that already? Honestly build vs new doesn't matter a ton, but also I'm pretty sure reply_order should auto fill, and if it doesn't that seems like a Thing to Fix?

Reply:
it looks like it doesn't set the reply_order until it/the post actually saves, which presumably causes problems in preview?